File: 268.missing-number.java
/*
 * @lc app=leetcode id=268 lang=java
 *
 * [268] Missing Number
 */

// @lc code=start
class Solution {
    public int missingNumber(int[] nums) {
        long sum = nums.length * (nums.length + 1) / 2;
        for (int n : nums) {
            sum -= n;
        }
        return (int) sum;
    }
}
// @lc code=end

File: 268.missing-number.cpp
/*
 * @lc app=leetcode id=268 lang=cpp
 *
 * [268] Missing Number
 */

// @lc code=start
#include <algorithm>
#include <iterator>
#include <vector>

using namespace std;

class Solution {
public:
    int missingNumber(vector<int>& nums) {
        vector<bool> bitmap(nums.size(), false);
        for (const auto &el : nums) bitmap[el] = true;
        return distance(bitmap.cbegin(), find(bitmap.cbegin(), bitmap.cend(), false));
    }
};
// @lc code=end
